File: ClientApp/src/app/interfaces/product/optionDto.ts
import { MultipleImageDto } from "./UserInteract/multipleImageDto";

export interface OptionDto {
    id: number | null;
    title: string;
    description: string | null;
    price: number;
    quantityAvailable: number;
    color: string | null;
    size: number | null;
    weight: number;
    genderFor: number | null;
    material: number | null;
    condition: number;
    images: MultipleImageDto[] | null;
}

File: ClientApp/src/app/interfaces/Product/OptionDto.cs
using Contracts.Model.Product.UserInteract;

namespace Contracts.Model.Product;

public class OptionDto
{
	public string Title { get; set; }
	public string? Description { get; set; }
	public double Price { get; set; }
	public int QuantityAvailable { get; set; }
	public string? Color { get; set; }
	public bool? IsMale { get; set; }
	public int Condition { get; set; }
	public ICollection<MultipleImageDto>? Images { get; set; }
}